core/capabilities/webapi/trigger/trigger.go

```diff
@@ -164,13 +171,62 @@ func (h *triggerConnectorHandler) HandleGatewayMessage(ctx context.Context, gate
 
 	default:
 		h.lggr.Errorw("unsupported method", "id", gatewayID, "method", body.Method)
-		err = h.sendResponse(ctx, gatewayID, body, ghcapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: fmt.Errorf("unsupported method %s", body.Method).Error()})
+		err = h.sendResponse(ctx, gatewayID, body, ghcapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "unsupported method " + body.Method})
 		if err != nil {
 			h.lggr.Errorw("error sending response", "err", err)
 		}
 	}
 }
 
+// Periodically update the gateways with the state of the workflow triggers.
+func (h *triggerConnectorHandler) loop(ctx context.Context) {
+	defer h.wg.Done()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Duration(updateGatewayIntervalS) * time.Millisecond):
+			err := h.updateGateways(ctx)
+			if err != nil {
+				h.lggr.Errorw("error updating gateways", "err", err)
+			}
+		}
+	}
+}
+
+// helper function to update gateways with the state of the workflow triggers
+func (h *triggerConnectorHandler) updateGateways(ctx context.Context) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	var workflowConfigs = make(map[string]*values.Map)
+	for triggerID, trigger := range h.registeredWorkflows {
+		workflowConfigs[triggerID] = trigger.rawConfig
+	}
+	payloadJSON, err := json.Marshal(workflowConfigs)
+	if err != nil {
+		h.lggr.Errorw("error marshalling payload", "err", err)
+		// TODO: Should we return here instead?
+		payloadJSON, _ = json.Marshal(ghcapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: fmt.Sprintf("error %s marshalling payload", err.Error())})
```

Comment:
I think we can log the error

Reply:
should we have a way to differentiate regular trigger request/response and metadata exchange request/response? e.g. introduce a new struct for metadata exchange or a new type to the existing request/response structs